corehq/apps/app_manager/translations.py

```diff
@@ -406,30 +406,47 @@ def update_case_list_translations(sheet, rows, app):
     module_index = int(sheet.worksheet.title.replace("module", "")) - 1
     module = app.get_module(module_index)
 
-    # It is easier to process the translations if mapping rows are nested under
-    # their respective DetailColumns
+    # It is easier to process the translations if mapping and graph config
+    # rows are nested under their respective DetailColumns.
+
     condensed_rows = []
-    i = 0
-    while i < len(rows):
-        if rows[i]['case_property'].endswith(" (ID Mapping Text)"):
-            # Cut off the id mapping text
-            rows[i]['case_property'] = rows[i]['case_property'].split(" ")[0]
-            # Construct a list of mapping rows
-            mappings = []
-            j = 1
-            while (i + j < len(rows) and
-                    rows[i + j]['case_property'].endswith(" (ID Mapping Value)")):
-                # Cut off the id mapping value part
-                rows[i + j]['case_property'] = \
-                    rows[i + j]['case_property'].split(" ")[0]
-                mappings.append(rows[i + j])
-                j += 1
-            rows[i]['mappings'] = mappings
-            condensed_rows.append(rows[i])
-            i += j
+    index_of_last_enum_in_condensed = -1
+    index_of_last_graph_in_condensed = -1
+    for i, row in enumerate(rows):
+        # If it's an enum case property, set index_of_last_enum_in_condensed
+        if row['case_property'].endswith(" (ID Mapping Text)"):
```

Comment:
I think my only remaining critique is that the strings used in this section (like `" (ID Mapping Text)"`) seem a bit fragile. They look like human-readable descriptions, and human readable things tend to get tweaked for readability. I'm worried that such tweaking at some remote part of the system could cause this code to break. A possibly better approach is to somehow tag each case property with a domain-specific name, using the syntax of a variable name. Typically someone changing such a name would do a search on the codebase first to make sure nothing else references that name.